sound/soc/sof/intel/hda-dsp.c

```diff
@@ -406,6 +409,11 @@ static int hda_dsp_update_d0i3c_register(struct snd_sof_dev *sdev, u8 value)
 	}
 
 	reg = snd_sof_dsp_read8(sdev, HDA_DSP_HDA_BAR, chip->d0i3_offset);
+	if ((reg & SOF_HDA_VS_D0I3C_I3) != value) {
+		dev_err(sdev->dev, "failed to update D0I3C!\n");
+		return -EIO;
```

Comment:
@plbossart yes this possible and this is exactly what happened in before this patch https://github.com/thesofproject/linux/pull/4173. I agree its because we messed up. But we'd have caught this mess if we had a check that said set D0i3C_I3 to 1 and read if it is 1 after CIP is cleared.

Reply:
ok, so in that case there should be a comment that this is a paranoia check to detect errors in programming sequences, this is not something that the hardware can do in the nominal case.